### ISSUE TITLE
Confusion about paper and metrics

### ISSUE THREAD

**Issue by KJKR73** [dpaleka/stealing-part-lm-supplementary#3]
Hi @dpaleka,

I had a great time reading your paper. Could you clarify some of my questions below. Would be very helpful.

1. How were the Bits of precision calculated and is it calculated on a logit level or log_prob or logit_softmax level?
2. The transformation of _hxh_, the h value here is the one that is calculated after hidden dimensionality attack if performed?

Thanks

**Comment by dpaleka** [comment]
> How were the Bits of precision calculated and is it calculated on a logit level or log_prob or logit_softmax level?

Logit (or logprobs, for bits of precision calculation it should be same). The code to compute this is unfortunately not in this repository, and the exact results cannot be reproduced without the black-box model logits we could not release. I expect the general trend to be mostly reproducible on another API or open-source models.

> The transformation of hxh, the h value here is the one that is calculated after hidden dimensionality attack if performed?

Yes, we determine this _h_ value from the singular value gap as described in the paper. For all open-source models we tested, except GPT-2, it matches the dimension up to +-1.

**Comment by KJKR73** [comment]
Hi @dpaleka, thanks for the answer,

I am calculating the bits of precision in this way, is it what you did as well ?

```
# Define the metrics and get the precision bits
y_pred = torch.softmax(recovered_logits, dim=-1).cpu()
y_true = torch.softmax(true_logits, dim=-1).cpu()
print(f"Bits of precision: {round(math.log2(mean_absolute_error(y_true, y_pred)), ndigits=1)}")
```

Thanks again :)

**Comment by carlini** [comment]
Let me clarify your question around logprobs. You want to know for Table 4 how we calculate the rows?

In which case, no, it's in logit space. So take log_softmax, and then calculate bits of precision as yeah basically what you write there.

**Comment by KJKR73** [comment]
I see thanks for clearing out the confusion :)
